DataFormats/Detectors/TRD/include/DataFormatsTRD/RawDataStats.h

```diff
@@ -0,0 +1,130 @@
+  uint64_t mTrackletWordsRead;                // tracklet words read in
+  uint64_t mTrackletWordsSkipped;             // tracklet words skipped for various reasons.
+  std::array<uint8_t, 1080> mLinkErrorFlag{}; //status of the error flags for this event, 8bit values from cru halfchamber header.
+};
+
+class TRDDataCountersPerTimeFrame
+{ //thisis on a per event basis
+ public:
+  std::array<uint32_t, 1080> mLinkNoData;                                   // Link had no data or was not present.
+  std::array<uint32_t, 1080> mLinkWords{};                                  //units of 256bits, read from the cru half chamber header
+  std::array<uint32_t, 1080> mLinkWordsRead{};                              // units of 32 bits the data words read before dumping or finishing
+  std::array<uint32_t, 1080> mLinkWordsDumped{};                            // units of 32 bits the data dumped due to some or other error
+  std::array<int64_t, o2::trd::constants::MAXMCMCOUNT> mLinkMCMsWithData{}; // and its corresponding volume of data.
+  std::array<uint32_t, constants::MAXMCMCOUNT> mMCMDigitCount{};
+  std::array<uint32_t, constants::MAXMCMCOUNT> mMCMTrackletCount{};
+  std::array<uint32_t, 30> mParsingErrors{};              // errors in parsing, indexed by enum above of ParsingErrors
+  std::array<uint32_t, 1080 * 30> mParsingErrorsByLink{}; // errors in parsing, indexed by enum above of ParsingErrors
+  uint64_t mTimeTaken;                                    // time taken to process the entire timeframe [ms].
+  uint64_t mTimeTakenForDigits;                           // time take to process tracklet data blocks [us].
+  uint64_t mTimeTakenForTracklets;                        // time take to process digit data blocks [us].
+  uint64_t mDigitsFound;                                  // digit found in the time frame.
+  uint64_t mTrackletsFound;                               // tracklets found in the time frame.
+  uint64_t mDigitWordsRead;                               // digit words read in.
+  uint64_t mDigitWordsSkipped;                            // digit words skipped for various reasons.
+  uint64_t mTrackletWordsRead;                            // tracklet words read in.
+  uint64_t mTrackletWordsSkipped;                         // tracklet words skipped for various reasons.
+  uint64_t mDataWordsRead;
+  uint64_t mDataWordsRejected;
+  //TRDDataCountersPerTimeFrame* operator=(TRDDataCountersPerTimeFrame *old){this=old;return *this;}
+};
+
+//TODO not sure this class is needed
+class TRDDataCountersRunning
+{                                                //those counters that keep counting
+  std::array<uint32_t, 1080> mLinkFreq{};        //units of 256bits "cru word"
+  std::array<bool, 1080> mLinkEmpty{};           // Link only has padding words only, probably not serious.
+  std::array<uint64_t, 65535> mDataFormatRead{}; // 7bits.7bits major.minor version read from HCHeader.
+};
+
```

Comment:
I think a list (ntuple) of format errors would be nice. The structure would essentially be the enum value from above, HCID, rob/mcm and channel where appropriate.
But this would need much more processing, so it should be part of a bigger PR.

Reply:
Ja that is along the lines of what i said in the meeting and Stefan asked for.